utils/getFeed.js

```diff
@@ -0,0 +1,151 @@
+	}
+
+	return items;
+}
+
+/**
+ * Returns a single feed item from a snapshot.
+ *
+ * @param {Document}        parsed     A parsed XML document.
+ * @param {Node}            snapshot   A snapshot returned from the snapshotItem method of a XPathResult.
+ * @param {XPathNSResolver} nsResolver A namespace resolver for the parsed document.
+ *
+ * @returns {FeedItem} The FeedItem representing the provided snapshot.
+ */
+function getFeedItem( parsed, snapshot, nsResolver ) {
+	let item     = {};
+
+	item.title       = getXPathText( 'child::title', parsed, snapshot );
+	item.link        = getXPathText( 'child::link', parsed, snapshot );
+	item.content     = getXPathText( 'child::content:encoded', parsed, snapshot, nsResolver );
+	item.description = getXPathText( 'child::description', parsed, snapshot );
+	item.creator     = getXPathText( 'child::dc:creator', parsed, snapshot, nsResolver );
+	item.date        = getXPathText( 'child::pubDate', parsed, snapshot );
+
+	return item;
+}
+
+/**
+ * Returns the string contents of the given xpath query on the provided document.
+ *
+ * @param {string}          xpath      The xpath query to run.
+ * @param {Document}        document   A parsed XML document.
+ * @param {Node}            context    A Node in the document to use as context for the query.
+ * @param {XPathNSResolver} nsResolver A namespace resolver for the parsed document.
+ *
+ * @returns {string|undefined} The string result of the xpath query.
+ */
+function getXPathText( xpath, document, context = null, nsResolver = null ) {
+	let result = document.evaluate( xpath , ( context || document ), nsResolver, XPathResult.STRING_TYPE, null );
+
+	if ( result.stringValue ) {
+		return result.stringValue;
+	}
+
+	return undefined;
+}
+
+/**
+ * Returns a ORDERED_NODE_SNAPSHOT_TYPE XpathResult for the given xpath query on the provided document.
+ *
+ * @param {string}          xpath      The xpath query to run.
+ * @param {Document}        document   A parsed XML document.
+ * @param {Node}            context    A Node in the document to use as context for the query.
+ * @param {XPathNSResolver} nsResolver A namespace resolver for the parsed document.
+ *
+ * @returns {XPathResult} The result of the xpath query.
+ */
+function getXPathSnapshots( xpath, document, context = null, nsResolver = null ) {
+	return document.evaluate( xpath , ( context || document ), nsResolver, XPathResult.ORDERED_NODE_SNAPSHOT_TYPE, null );
+}
+
+/**
+ * Grabs an RSS feed from the requested URL and parses it.
+ *
+ * @param {string} url      The URL the feed is located at.
+ * @param {int}    maxItems The amount of items you wish returned, 0 for all items.
+ *
+ * @returns {Promise.<Feed>} The retrieved feed.
+ */
+export default function getFeed( url, maxItems = 0 ) {
+	return fetch( url ).then( function ( response ) {
+		return response.text();
+	} ).then( function ( raw ) {
+		return parseFeed( raw, maxItems );
+	} );
```

Comment:
No catch clause.

Reply:
Regardless of any catch clauses a failed CORS request will always log an error in the console. This can't be avoided.

If specific error handling is required a catch method can be chained to the `getFeed` function.

I did change `parseFeed` to also return a promise with the appropriate `reject` call for errors in case a feed returns something that can't be parsed as XML.